### ISSUE TITLE
bitnami/minio is not starting correctly in github actions

### ISSUE THREAD

**Issue by jwoodrow** [bitnami/containers#30684]
### Name and Version

bitnami/minio:2023.3.24

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. Create a github action using the workflow

```yaml
# ...
jobs:
  job:
    # ...
    services:
      mooncard-minio:
        image: bitnami/minio:2023.3.24
        env:
          MINIO_ROOT_USER: ROOT_USER
          MINIO_ROOT_PASSWORD: ROOT_PASSWORD
        options: >-
          --health-cmd "curl -I http://localhost:9000/minio/health/live"
          --health-interval 10s
          --health-timeout 5s
          --health-retries 6
          --health-start-period 30s
# ...
```

2. run the workflow

### What is the expected behavior?

The Minio service start fine on the default ports (API 9000 | Console 9001)

### What do you see instead?

```bash
/usr/bin/docker logs --details f8ef195ed148728103ac2d0f7ed9fd23e0f0560d8781c5ab2556b78dfc23aa4b
    13:05:08.55 
    13:05:08.55 Welcome to the Bitnami minio container
    13:05:08.55 Subscribe to project updates by watching https://github.com/bitnami/containers
    13:05:08.55 Submit issues and feature requests at [https://github.com/bitnami/containers/issues](https://github.com/bitnami/containers/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)
    13:05:08.55 
    13:05:08.55 INFO  ==> ** Starting MinIO setup **
   minio 13:05:08.57 INFO  ==> Starting MinIO in background...
   minio 13:05:13.59 INFO  ==> Adding local Minio host to 'mc' configuration...
   minio 13:05:13.70 INFO  ==> Stopping MinIO...
    13:05:13.81 INFO  ==> ** MinIO setup finished! **
   minio 13:05:13.82 INFO  ==> ** Starting MinIO **
   
   MinIO Object Storage Server
   Copyright: 2015-2023 MinIO, Inc.
   License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
   Version: DEVELOPMENT.2023-03-24T21-41-23Z (go1.19.7 linux/amd64)
   
   Status:         1 Online, 0 Offline. 
   API: http://localhost:9000 
   Console: http://172.18.0.4:9001/ http://127.0.0.1:9001/ 
   
   Documentation: https://min.io/docs/minio/linux/index.html
   Warning: The standard parity is set to 0. This can lead to data loss.
   ERROR Unable to initialize console server: Specified port is already in use
         > Please ensure no other program uses the same address/port
  Error: Failed to initialize container bitnami/minio
```

### Additional information

Locking the version to `2023.3.22` seems to resolve the issue completely

**Comment by gongomgra** [comment]
Hi @jwoodrow,

Thanks for using Bitnami and for reporting this. I'm trying to reproduce your issue on my side and I will keep you posted. I will move the ticket to the "on hold" column for the stale bot not to automatically close the ticket.

**Comment by gongomgra** [comment]
Hi @jwoodrow,

Sorry for the delay. I'm afraid I can't reproduce your issue in a custom GitHub action I created for this. Mine only deploys the minio service binding the ports to the host, and then it runs a curl command against the service to verify it is alive (apart from the health check already configured in the service itself). Here is my GitHub action file for further reference:

```
# This is a basic workflow to help you get started with Actions

name: CI

# Controls when the workflow will run
on:
  # Triggers the workflow on push or pull request events but only for the "main" branch
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest
    services:
      minio:
        # image: bitnami/minio:2023.3.24
        image: bitnami/minio:2023.3.24-debian-11-r6
        env:
          MINIO_ROOT_USER: admin
          MINIO_ROOT_PASSWORD: bitnami123
          BITNAMI_DEBUG: true
        options: >-
          --health-cmd "curl -I http://localhost:9000/minio/health/live"
          --health-interval 10s
          --health-timeout 5s
          --health-retries 6
          --health-start-period 30s
        ports:
          - 9000:9000
          - 9001:9001

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Runs a single command using the runners shell
      - name: Run a one-line script
        run: echo Hello, world!

      # Runs a set of commands using the runners shell
      - name: Run a multi-line script
        run: |
          echo "Run a curl-based health command"
          curl -LI http://localhost:9000/minio/health/live
          echo "exitCode: $?"
```

It worked just fine for me on every run (I ran it 5 times for testing purposes). I set the `BITNAMI_DEBUG: true` environment variable to get more information in case of errors. Can you set it on your action to try to get more information about the error?

@arnovdk, I think your error can be related to the group `1001` that you are setting in the `user: 1001:1001` line (and the `PGID` value). Our non-root containers requires the group to be set to `0` (zero) or `root` as you can see in the comment linked below

https://github.com/bitnami/charts/issues/16102#issuecomment-1516361649

@jwoodrow you didn't share the rest of your GitHub action file, but can it be related to your issue as well? Are you setting the non-root UID/GID values for the Minio container?

**Comment by jwoodrow** [comment]
> @jwoodrow you didn't share the rest of your GitHub action file, but can it be related to your issue as well? Are you setting the non-root UID/GID values for the Minio container?

Our main job (the one running the tests) is in fact running in a container with a non-root user. But the daemon itself is not rootless so I don't think this should impact the minio service starting. I'll try and find some time to check with `BITNAMI_DEBUG: true` on the minio service and get back to you.

I don't think the problem would come from the rest of my GH action workflow since it doesn't even reach the job step and would just fail whilst initializing the services.
Nothing was changed either (it broke once the `2023.3.24` version was released because we didn't version lock the version for the service) and once we locked the version to the prior release everything worked as before.

**Comment by arnovdk** [comment]
Thank you @jwoodrow , I can confirm that (in my case) changing the value of `GID` to `0` solved my problems. Both of the mentioned errors disappeared. Thanks a bunch!

I must confess that I don't really recall how I got to using the value `1001` for `GID`, but apparently I had been using this config for a while now (since I started, some time last year) and things finally broke since release 2023.3.24-debian-11-r6.

Presently I am using just `bitnami/minio` without tag and things the service starts up again. Thanks again! 🙂

Note: obviously I found this issue because I was doing a search towards the error text that I got and therefore I thought both situations where similar.

**Comment by gongomgra** [comment]
Thanks for your messages. I'm moving the ticket to solved.
